paint_house.py

```diff
@@ -0,0 +1,19 @@
+#Tume complexity: O(n): where n: # of houses
+#Space complexity: O(n)
```

Comment:
Shouldn't your SC be O(1)? You haven't used any extra space. You just solved this question in-place, which in my opinion, you shouldn't have.